cljfmt/src/cljfmt/main.clj

```diff
@@ -0,0 +1,189 @@
+   :remove-consecutive-blank-lines? true
+   :indents cljfmt/default-indents
+   :alias-map {}})
+
+(defn- abort
+  [& msg]
+  (binding [*out* *err*]
+    (when (seq msg)
+      (apply println msg))
+    (System/exit 1)))
+
+(defn- warn
+  [& args]
+  (binding [*out* *err*]
+    (apply println args)))
+
+(defn- relative-path [dir file]
+  (-> (.toURI dir)
+      (.relativize (.toURI file))
+      (.getPath)))
+
+(defn- grep [re dir]
+  (filter #(re-find re (relative-path dir %)) (file-seq (io/file dir))))
+
+(defn- find-files [{:keys [file-pattern]} f]
+  (let [f (io/file f)]
+    (when-not (.exists f) (abort "No such file:" (str f)))
+    (if (.isDirectory f)
+      (grep file-pattern f)
+      [f])))
+
+(defn- reformat-string [options s]
+  (cljfmt/reformat-string s options))
+
+(defn- project-path [{:keys [project-root]} file]
+  (-> project-root io/file (relative-path (io/file file))))
+
+(defn- format-diff
+  ([options file]
+   (let [original (slurp (io/file file))]
+     (format-diff options file original (reformat-string options original))))
+  ([options file original revised]
+   (let [filename (project-path options file)
+         diff     (diff/unified-diff filename original revised)]
+     (if (:ansi? options)
+       (diff/colorize-diff diff)
+       diff))))
+
+(def ^:private zero-counts {:okay 0, :incorrect 0, :error 0})
+
+(defn- check-one [options file]
+  (let [original (slurp file)
+        status   {:counts zero-counts :file file}]
+    (try
+      (let [revised (reformat-string options original)]
+        (if (not= original revised)
+          (-> status
+              (assoc-in [:counts :incorrect] 1)
+              (assoc :diff (format-diff options file original revised)))
+          (assoc-in status [:counts :okay] 1)))
+      (catch Exception ex
+        (-> status
+            (assoc-in [:counts :error] 1)
+            (assoc :exception ex))))))
+
+(defn- print-stack-trace [ex]
+  (binding [*out* *err*]
+    (st/print-stack-trace ex)))
+
+(defn- print-file-status [options status]
+  (let [path (project-path options (:file status))]
+    (when-let [ex (:exception status)]
+      (warn "Failed to format file:" path)
+      (print-stack-trace ex))
+    (when-let [diff (:diff status)]
+      (warn path "has incorrect formatting")
+      (warn diff))))
+
+(defn- exit [counts]
+  (when-not (zero? (:error counts 0))
+    (System/exit 2))
+  (when-not (zero? (:incorrect counts 0))
+    (System/exit 1)))
+
+(defn- print-final-count [counts]
+  (let [error     (:error counts 0)
+        incorrect (:incorrect counts 0)]
+    (when-not (zero? error)
+      (warn error "file(s) could not be parsed for formatting"))
+    (when-not (zero? incorrect)
+      (warn incorrect "file(s) formatted incorrectly"))
+    (when (and (zero? incorrect) (zero? error))
+      (println "All source files formatted correctly"))))
+
+(defn- merge-counts
+  ([]    zero-counts)
+  ([a]   a)
+  ([a b] (merge-with + a b)))
+
+(defn check
+  [options paths]
```

Comment:
Ideally `options` should come after `paths`, and we should also allow a one-arity form, e.g. `(check paths)`.